In [1]:
import numpy as np
import scipy.stats as stats

1-4 задачи решать с помощью функций, 5ю вручную<br>
Выбрать тест и проверить, есть  ли различия между выборками:<br>

**Задача 1.** Даны две  независимые выборки. Не соблюдается условие нормальности<br>
x1  380,420, 290<br>
y1 140,360,200,900<br>

Сделайте вывод по результатам, полученным с помощью функции<br>

Для решения данной задачи выбираем критерий Манна-Уитни

In [5]:
alpha = 0.05
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

In [6]:
stats.mannwhitneyu(x1, y1)


MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

**Ответ:** Так как pvalue > alpha, то это дает нам право сделать выбор в пользу нулевой гипотезы H0, статистически значимых отличий нет.

**Задача 2.**  Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?<br>

1е измерение до приема препарата: 150, 160, 165, 145, 155<br>
2е измерение через 10 минут: 140, 155, 150,  130, 135<br>
3е измерение через 30 минут: 130, 130, 120, 130, 125<br>

Для решения данной задачи выбираем критерий Фридмана

In [9]:
alpha = 0.05
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])
x3 = np.array([130, 130, 120, 130, 125])

In [11]:
stats.friedmanchisquare(x1, x2, x3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

**Ответ:** Так как pvalue < alpha, то это дает нам право сделать выбор в пользу H1 гипотезы, H0 гипотеза отвергается, есть статистически значимые отличия

**Задача 3.** Сравните 1 и 2е измерения, предполагая, что 3го измерения через 30 минут не было.

Для решения данной задачи выбираем критерий Уилкоксона

In [16]:
alpha = 0.05
x1 = np.array([150, 160, 165, 145, 155])
x2 = np.array([140, 155, 150, 130, 135])

In [17]:
stats.wilcoxon(x1, x2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

**Ответ:** Так как pvalue > alpha, то это дает нам право сделать выбор в пользу нулевой гипотезы H0, статистически значимых отличий нет.

**Задача 4.** Даны 3 группы  учеников плавания.<br>
В 1 группе время на дистанцию 50 м составляют:<br>
56, 60, 62, 55, 71, 67, 59, 58, 64, 67<br>

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53<br>
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54<br>


Для решения данной задачи выбираем критерий Крускала-Уоллиса

In [18]:
alpha = 0.05
x1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
x2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
x3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [19]:
stats.kruskal(x1, x2, x3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

**Ответ:** Так как pvalue > alpha, то это дает нам право сделать выбор в пользу нулевой гипотезы H0, статистически значимых отличий нет.

**Задача 5.** Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону распределения. Объем выборки 10, уровень статистической значимости 5%.<br>

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [31]:
X = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
alpha = 0.05
n = 10
mu = 2.5

Так как неизвестна $\sigma$ генеральной совокупности, то для проверки гипотезы будем использовать t-критерий: <br>
$$ t = \dfrac{Xср - μ}{\sigmaн / \sqrt{n}} $$

In [33]:
mean = np.mean(X)
print(mean)

2.5279999999999996


In [36]:
sigma = np.std(X,ddof=1)
print(sigma)

0.1572542173961923


In [41]:
t_X = (mean - mu)/(sigma/np.sqrt(n))
print(t_X)

0.5630613661802959


Табличное значение t-критерия  tтабл.= 2.2622, расчётное tрасч. = 0.56

**Ответ:** Так как tтабл. > tрасч., то принимаем нулевую гипотезу H0